src/ripple/rpc/impl/NFTokenID.cpp

```diff
@@ -0,0 +1,200 @@
+                [](STObject const& nft) {
+                    return nft.getFieldH256(sfNFTokenID);
+                });
+        }
+        else if (fName == sfModifiedNode)
+        {
+            // When a mint results in splitting an existing page,
+            // it results in a created page and a modified node. Sometimes,
+            // the created node needs to be linked to a third page, resulting
+            // in modifying that third page's PreviousPageMin or NextPageMin
+            // field changing, but no NFTs within that page changing. In this
+            // case, there will be no previous NFTs and we need to skip.
+            // However, there will always be NFTs listed in the final fields,
+            // as rippled outputs all fields in final fields even if they were
+            // not changed.
+            STObject const& previousFields =
+                node.peekAtField(sfPreviousFields).downcast<STObject>();
+            if (!previousFields.isFieldPresent(sfNFTokens))
+                continue;
+
+            STArray const& toAddPrevNFTs =
+                previousFields.getFieldArray(sfNFTokens);
+            std::transform(
+                toAddPrevNFTs.begin(),
+                toAddPrevNFTs.end(),
+                std::back_inserter(prevIDs),
+                [](STObject const& nft) {
+                    return nft.getFieldH256(sfNFTokenID);
+                });
+
+            STArray const& toAddFinalNFTs = node.peekAtField(sfFinalFields)
+                                                .downcast<STObject>()
+                                                .getFieldArray(sfNFTokens);
+            std::transform(
+                toAddFinalNFTs.begin(),
+                toAddFinalNFTs.end(),
+                std::back_inserter(finalIDs),
+                [](STObject const& nft) {
+                    return nft.getFieldH256(sfNFTokenID);
+                });
+        }
+    }
+
+    // We expect NFTs to be added one at a time.  So finalIDs should be one
+    // longer than prevIDs.  If that's not the case something is messed up.
+    if (finalIDs.size() != prevIDs.size() + 1)
+        return std::nullopt;
+
+    // Find the first NFT ID that doesn't match.  We're looking for an
+    // added NFT, so the one we want will be the mismatch in finalIDs.
+    auto const diff = std::mismatch(
+        finalIDs.begin(), finalIDs.end(), prevIDs.begin(), prevIDs.end());
+
+    // There should always be a difference so the returned finalIDs
+    // iterator should never be end().  But better safe than sorry.
+    if (diff.first == finalIDs.end())
+        return std::nullopt;
+
+    return *diff.first;
+}
+
+static std::vector<uint256>
+getNFTokenIDFromDeletedOffer(TxMeta const& transactionMeta)
+{
+    std::vector<uint256> tokenIDResult;
+    for (STObject const& node : transactionMeta.getNodes())
+    {
+        if (node.getFieldU16(sfLedgerEntryType) != ltNFTOKEN_OFFER)
+            continue;
```

Comment:
you should do:

```c++
if (node.getFieldU16(sfLedgerEntryType) != ltNFTOKEN_OFFER || node.getFName() != sfDeletedNode)
  continue;
```

so that you don't need an additional conditional below